packages/jest-environment-jsdom/src/index.ts

```diff
@@ -102,12 +98,16 @@ class JSDOMEnvironment {
     return Promise.resolve();
   }
 
-  runScript(script: Script): ?any {
+  runScript(script: Script) {
     if (this.dom) {
-      return this.dom.runVMScript(script);
+      // `runVMScript` returns `void` and JestEnvironment.runScript is expected
+      // to return `... | null` so we don't return (`void`) early here:
+      //
+      // return this.dom.runVMScript(script);
```

Comment:
This is wrong, you _must_ return the result of the evaluation. If the typing says that this is `void` they are wrong - `runVMScript` returns the result of evaluating the script passed in.

```js
const {Script} = require('vm');
const {JSDOM} = require('jsdom');

const dom = new JSDOM('', {runScripts: 'dangerously'});
const script = new Script('42');

const value = dom.runVMScript(script);

console.log(value === 42); // true
```

Reply:
the typing should either return `any` or `unknown` - it cannot statically know the return type